spring-cloud-gcp-docs/src/main/asciidoc/logging.adoc

```diff
@@ -114,7 +114,16 @@ The following properties are available:
 [options="header",]
 |=======================================================================
 | Property | Default Value | Description
-| `projectId` |  | Only required for logging the correct traceId: `projects/[PROJECT-ID]/traces/[TRACE-ID]`. If `projectId` is not set, the logged `traceId` will match the one provided by `spring-cloud-gpc-trace`
+| `projectId`
+a| If not set, default value is determined in the following order:
+
+1. `SPRING_CLOUD_GCP_LOGGING_PROJECT_ID` Environmental Variable.
```

Comment:
I'm not really sure about this since, as I mentioned earlier, if a user wants auto-configuration, why don't they use it explicitly?

Reply:
The ideal situation is to consume Spring configurations from Layout. But it seems that the Layout is instantiated out of band, and has no access to the context.

Reply:
I don't know what the layout is, but I'm assuming you're referring to something static/compile time?

But weren't we auto-configuring these before? Was the auto-configuration not being used then?

Reply:
The configuration is in the Logback Layout, which is initialized outside of Spring. I.e., it's initialized by Logback:
https://github.com/saturnism/spring-cloud-gcp/blob/ee0b421da3aaa9468286fca54b9e6fd01b5301a7/spring-cloud-gcp-autoconfigure/src/main/resources/org/springframework/cloud/gcp/autoconfigure/logging/logback-json-appender.xml#L10

Thus it has no access to the Spring context.

Reply:
It turns out we weren't even using the project ID in auto-configuration before.
Should we omit the `If not set, default value is ` bit since, even if users specify a project ID through a property, it won't be used anywhere?